### DISCUSSION TITLE
Advanced Candid argument operations

### DISCUSSION THREAD

**Discussion by nmattia** [dfinity/sdk#2623] [Feature Requests]
Currently (AFAICT) I can only pass full-fledged arguments to `dfx install`, e.g.:

``` bash
$ dfx canister install --argument '(record {foo="bar"})'
```

I would like dfx to be able to read a (CLI) argument and splice it in for me, so that I don't have to deal with any shell escaping madness:

```
$ dfx canister install --value bar baz --argument '(record {foo=$bar})'
# install argument expands to '(record {foo="baz"})'
```

See for instance the various options e.g. `jq` offers for setting arguments/values:

```
$ jq --help
...
  --arg a v        set variable $a to value <v>;
  --argjson a v    set variable $a to JSON value <v>;
  --slurpfile a f  set variable $a to an array of JSON texts read from <f>;
  --rawfile a f    set variable $a to a string consisting of the contents of <f>;
```

**Comment by ghost** [comment]
The example above has `--value foo bar` on the command-line, seemingly identifying the field name `foo` and the variable name `bar`, but not the variable value.

First question: would this be a better example?
```
$ dfx canister install --value bar baz --argument '(record {foo=$bar})'
# install argument expands to '(record {foo="baz"})'
```

**Reply by nmattia** [reply]
Yes, thanks! Edited
